### PR TITLE
[state-transitions] handle completing an incomplete declaration

### PR DESCRIPTION
**Pull request by Zangetsu101**

### CODE REVIEW

File: packages/gateway/src/features/registration/type-resolvers.ts [hunk 1]
```diff
@@ -1640,7 +1640,13 @@ export const typeResolvers: GQLResolver = {
     },
     async _fhirIDMap(record: Saved<Bundle>) {
       const composition = getComposition(record)
-      const encounter = getEncounterFromRecord(record, DEATH_ENCOUNTER_CODE)
+      const encounter = findEncounterFromRecord(record, DEATH_ENCOUNTER_CODE)
+
+      if (!encounter) {
+        return {
+          composition: composition.id
```
Comment (tahmidrahman-dsi) [inline packages/gateway/src/features/registration/type-resolvers.ts:1647]: Can there be `questionnaireResponse` without encounter in `fhirIDMap`?

Comment (Zangetsu101) [reply packages/gateway/src/features/registration/type-resolvers.ts:1647]: Good question. I guess we need to add all the values one by one based on availability

Comment (tahmidrahman-dsi) [reply packages/gateway/src/features/registration/type-resolvers.ts:1647]: I think that is a good idea

Comment (Zangetsu101) [reply packages/gateway/src/features/registration/type-resolvers.ts:1647]: it looks like that's not needed as Questionnaires & Observations are dependent on encounter

**Pull request merged**